evm_loader/program/src/evm/mod.rs

```diff
@@ -247,13 +247,13 @@ impl<B: Database> Machine<B> {
             context: Context {
                 caller: origin,
                 contract: target,
-                value: trx.value,
+                value: *trx.value(),
                 code_address: Some(target),
             },
-            gas_price: trx.gas_price,
-            gas_limit: trx.gas_limit,
+            gas_price: *trx.gas_price(),
+            gas_limit: *trx.gas_limit(),
             execution_code,
-            call_data: trx.call_data,
+            call_data: trx.call_data().clone(),
```

Comment:
`call_data()` can be extremely large. Don't clone it.

Reply:
rearranged datastructures to use mutable reference and std::mem::take